pandas/io/parsers.py

```diff
@@ -3169,7 +3170,7 @@ def _make_date_converter(date_parser=None, dayfirst=False,
                          infer_datetime_format=False):
     def converter(*date_cols):
         if date_parser is None:
-            strs = _concat_date_cols(date_cols)
+            strs = _concat_date_cols(date_cols, keep_trivial_numbers=True)
```

Comment:
when is this called w/o keep_trivial_numbers=True, why is this not the default?

Reply:
Because we tried to keep the original behaviour. That is, `_concat_date_cols` previously always returned a numpy array with strings inside, but for parsing it to datetimes it's unnecessary as parsing code already handles `int`-s and `float`-s just fine.

So we've added the flag that speeds concatenation _when we're sure_ it won't break stuff.